newIDE/app/src/Profile/Subscription/SubscriptionDialog.js

```diff
@@ -284,44 +318,127 @@ export default function SubscriptionDialog({
-              ) : (
-                <React.Fragment key="upgrade">
-                  <LeftLoader isLoading={isLoading}>
+                (!!userPlanId && userPlanId === plan.legacyPlanId) ||
+                // Education plans (yearly and monthly) are merged into a single plan in the UI
+                // with a selector.
+                (!!userPlanId &&
+                  !!plan.planId &&
+                  userPlanId.startsWith(plan.planId));
+              let actions: React.Node = null;
+              if (isFreePlan) {
+                // If no plan (free usage), do not display button.
+              } else if (plan.planId === 'gdevelop_education') {
+                if (!isUserCurrentOrLegacyPlan) {
+                  actions = [
+                    <ResponsiveLineStackLayout
+                      key="options"
+                      expand
+                      noColumnMargin
+                      noMargin
+                    >
+                      <SemiControlledTextField
+                        value={educationPlanSeatsCount.toString()}
+                        floatingLabelFixed
+                        fullWidth
+                        floatingLabelText={<Trans>Number of seats</Trans>}
+                        commitOnBlur
+                        type="number"
+                        style={{}}
+                        onChange={value => {
+                          const newValue = parseInt(value);
+                          setEducationPlanSeatsCount(
+                            Math.min(
+                              EDUCATION_PLAN_MAX_SEATS,
+                              Math.max(
+                                Number.isNaN(newValue)
+                                  ? EDUCATION_PLAN_MIN_SEATS
+                                  : newValue,
+                                EDUCATION_PLAN_MIN_SEATS
+                              )
+                            )
+                          );
+                        }}
+                        min={EDUCATION_PLAN_MIN_SEATS}
+                        max={EDUCATION_PLAN_MAX_SEATS}
+                        step={1}
+                        helperMarkdownText={i18n._(
+                          t`As a teacher, you will use one seat in the plan so make sure to include yourself!`
+                        )}
+                      />
+                      <SelectField
+                        value={educationPlanPeriodicity}
+                        floatingLabelText={<Trans>Engagement</Trans>}
+                        fullWidth
+                        onChange={(e, i, newValue) => {
+                          // $FlowExpectedError - Flow does not infer the type given the select options.
+                          setEducationPlanPeriodicity(newValue);
+                        }}
+                      >
+                        <SelectOption value="year" label={t`Per year`} />
+                        <SelectOption value="month" label={t`Per month`} />
```

Comment:
should the values be more `yearly` and `monthly`? (doesn't change much, just feels clearer)